### PR TITLE
Added a few functions to support TF ND Provider

### PR DESCRIPTION
**Pull request by sajagana**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -386,3 +410,61 @@ func stripQuotes(word string) string {
...
+	if method == "GET" {
+		req, err = http.NewRequest(method, fURL.String(), nil)
+	} else {
+		req, err = http.NewRequest(method, fURL.String(), bytes.NewBuffer(payload))
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if c.skipLoggingPayload {
```
Comment (akinross) [inline client/client.go:453]: https://github.com/ciscoecosystem/aci-go-client/pull/294

will this also impact us in this client?

**Review by akinross — CHANGES_REQUESTED** [review]

File: client/client.go (hunk 1 above)
Comment (lhercot) [reply client/client.go:453]: Most likely.

**Comment by lhercot** [comment]
Should we have a separate project for nd-go-client which only include the client part and can be streamlined for the generated provider? Should that client be a separate project or added to TF provider?

**Comment by akinross** [comment]
> Should we have a separate project for nd-go-client which only include the client part and can be streamlined for the generated provider? Should that client be a separate project or added to TF provider?

For me the separate project makes sense if this is purely the request handling towards ND which could also be used by other repositories like MSO. I do not foresee a lot of changes there, like we have had with model changes in past for aci-go-client and mso-go-client. Perhaps if we go that route, we should consider creating a generic REST request handler project for the various products? Not sure this is feasible and worth the time though, but we would be able to reuse certain  parts of code.

In case we will not have the dependency then I would keep the client inside of the repository, since that would make it easier to control changes in single PR.

**Comment by lhercot** [comment]
Moving this code into the ND provider repo directly. So closing this PR.

**Pull request closed without merging**